test/extended/cluster/cm.go

```diff
@@ -0,0 +1,189 @@
+			e2e.Failf("Error listing namespaces: %v\n", err)
+		}
+
+		// Crawl namespace slice
+		for _, ns := range nsList.Items {
+			var goodIndex int
+
+			if strings.Contains(ns.Name, "openshift") {
+				continue
+			}
+			// List objects in current namespace
+			e2e.Logf("Listing objects in namespace %v", ns.Name)
+
+			// Check for DeploymentConfigs
+			dcs, err := oc.AdminAppsClient().Apps().DeploymentConfigs(ns.Name).List(metav1.ListOptions{})
+			if err != nil {
+				e2e.Failf("Error listing DeploymentConfigs: %v\n", err)
+			}
+
+			for _, dc := range dcs.Items {
+				dc, err := oc.AdminAppsClient().Apps().DeploymentConfigs(ns.Name).Get(dc.Name, metav1.GetOptions{})
+				if err != nil {
+					e2e.Failf("Error DC not found: %v\n", err)
+				}
+				if templateName, ok := dc.Labels["template"]; ok {
+					// Get DC that has template label
+					nsTemplate[ns.Name] = make(map[string]struct{})
+					nsTemplate[ns.Name][templateName] = struct{}{}
+					e2e.Logf("DC found, with template label: %+v", templateName)
+				} else {
+					e2e.Logf("No template associated with this DC: %s", dc.Name)
+				}
+			}
+
+			// List pods in namespace
+			pods, err = c.CoreV1().Pods(ns.Name).List(metav1.ListOptions{})
+			if err != nil {
+				e2e.Failf("Error listing pods: %v\n", err)
+			}
+
+			for i, pod := range pods.Items {
+
+				// Only consider running pods
+				if pod.Status.Phase == v1.PodRunning {
+					// If the pod is part of a deployment we will take the template name instead
+					if value, ok := pod.Labels["deployment"]; ok {
+						// Get RC that matches pod label
+						rc, err := c.CoreV1().ReplicationControllers(ns.Name).Get(value, metav1.GetOptions{})
+						if err != nil {
+							e2e.Failf("Error RC not found: %v\n", err)
+						}
+						// Find template name from RC labels
+						if templateName, ok := rc.Labels["template"]; ok {
+							nsTemplate[ns.Name] = make(map[string]struct{})
+							nsTemplate[ns.Name][templateName] = struct{}{}
+							e2e.Logf("RC found, with template label: %+v", templateName)
+						} else {
+							e2e.Logf("No template associated with this RC: %s", rc.Name)
+						}
+					} else {
+						// Save standalone pods only
+						pods.Items[goodIndex] = pods.Items[i]
+						goodIndex++
+					}
+				}
+			}
+			nsPod[ns.Name] = pods.Items[:goodIndex]
+		}
+
+		// Crawl of namespace-pod map
+		for k, v := range nsPod {
+			if _, ok := excludedNS[k]; !ok {
+				ns := newNS(k)
+				for i := range v {
+					ns.Pods = append(ns.Pods, newPod(nsPod[k][i]))
+				}
+				for i := range nsTemplate[k] {
+					ns.Templates = append(ns.Templates, newTemplate(i))
+				}
+				if len(ns.Pods) > 0 || len(ns.Templates) > 0 {
+					config.ClusterLoader.Projects = append(config.ClusterLoader.Projects, ns)
+				}
+			}
+		}
+
+		// Marshal CL config struct to yaml
+		d, err := yaml.Marshal(&config)
+		if err != nil {
+			log.Fatalf("error: %v", err)
+		}
+
+		// Write cl.yaml file
+		err = ioutil.WriteFile("cl.yml", d, 0644)
```

Comment:
Could "cl.yml" be a constant somewhere at the top, rather than hard-coding it here?

Reply:
Agreed